packages/kbn-es-archiver/src/cli.ts

```diff
@@ -0,0 +1,244 @@
+const defaultConfigPath = resolveConfigPath('test/functional/config.js');
+
+export function runCli() {
+  new RunWithCommands({
+    description: 'CLI to manage archiving/restoring data in elasticsearch',
+    globalFlags: {
+      string: ['es-url', 'kibana-url', 'dir', 'config'],
+      help: `
+        --config           path to an FTR config file that sets --es-url, --kibana-url, and --dir
+                             default: ${defaultConfigPath}
+        --es-url           url for Elasticsearch, prefer the --config flag
+        --kibana-url       url for Kibana, prefer the --config flag
+        --dir              where arechives are stored, prefer the --config flag
+      `,
+    },
+    async extendContext({ log, flags, addCleanupTask }) {
+      const configPath = flags.config || defaultConfigPath;
+      if (typeof configPath !== 'string') {
+        throw createFlagError('--config must be a string');
+      }
+      const config = await readConfigFile(log, Path.resolve(configPath));
+
+      let esUrl = flags['es-url'];
+      if (esUrl && typeof esUrl !== 'string') {
+        throw createFlagError('--es-url must be a string');
+      }
+      if (!esUrl && config) {
+        esUrl = Url.format(config.get('servers.elasticsearch'));
+      }
+      if (!esUrl) {
+        throw createFlagError('--es-url or --config must be defined');
+      }
+
+      let kibanaUrl = flags['kibana-url'];
+      if (kibanaUrl && typeof kibanaUrl !== 'string') {
+        throw createFlagError('--kibana-url must be a string');
+      }
+      if (!kibanaUrl && config) {
+        kibanaUrl = Url.format(config.get('servers.kibana'));
+      }
+      if (!kibanaUrl) {
+        throw createFlagError('--kibana-url or --config must be defined');
+      }
+
+      let dir = flags.dir;
+      if (dir && typeof dir !== 'string') {
+        throw createFlagError('--dir must be a string');
+      }
+      if (!dir && config) {
+        dir = Path.resolve(config.get('esArchiver.directory'));
+      }
+      if (!dir) {
+        throw createFlagError('--dir or --config must be defined');
+      }
+
+      const client = new legacyElasticsearch.Client({
+        host: esUrl,
+        log: flags.verbose ? 'trace' : [],
+      });
+      addCleanupTask(() => client.close());
+
+      const esArchiver = new EsArchiver({
+        log,
+        client,
+        dataDir: dir,
+        kibanaUrl,
+      });
+
+      return {
+        esArchiver,
+      };
+    },
+  })
+    .command({
+      name: 'save',
+      usage: 'save [name] [...indices]',
+      description: `
+        archive the [indices ...] into the --dir with [name]
+
+        Example:
+          Save all [logstash-*] indices from http://localhost:9200 to [snapshots/my_test_data] directory
+
+          WARNING: If the [my_test_data] snapshot exists it will be deleted!
+
+          $ node scripts/es_archiver save my_test_data logstash-* --dir snapshots
```

Comment:
I tried to run it and got this error (I started OSS Kibana and loaded ecommerce sample data in advance)
```
➜  kibana git:(implement/es-archiver-package) node scripts/es_archiver.js save my_test_data kibana_sample_data_ecommerce --dir test/functional/fixtures/es_archiver/
ERROR UNHANDLED ERROR
ERROR TypeError: Cannot read property 'tasks' of undefined
          at add (/Users/dzmitrylemechko/github/kibana/packages/kbn-dev-utils/target/run/cleanup.js:48:14)
          at Object.extendContext (/Users/dzmitrylemechko/github/kibana/packages/kbn-es-archiver/src/cli.ts:96:7)
          at process._tickCallback (internal/process/next_tick.js:68:7)
          at Function.Module.runMain (internal/modules/cjs/loader.js:834:11)
          at startup (internal/bootstrap/node.js:283:19)
          at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)
```

Reply:
Same error with `load` command:

```
➜  kibana git:(implement/es-archiver-package) node scripts/es_archiver.js load test/functional/fixtures/es_archiver/date_nanos --config test/functional/config.js
ERROR UNHANDLED ERROR
ERROR TypeError: Cannot read property 'tasks' of undefined
          at add (/Users/dzmitrylemechko/github/kibana/packages/kbn-dev-utils/target/run/cleanup.js:48:14)
          at Object.extendContext (/Users/dzmitrylemechko/github/kibana/packages/kbn-es-archiver/src/cli.ts:96:7)
          at process._tickCallback (internal/process/next_tick.js:68:7)
          at Function.Module.runMain (internal/modules/cjs/loader.js:834:11)
          at startup (internal/bootstrap/node.js:283:19)
          at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)
```

I did bootstrap and can see line `succ [@kbn/es-archiver] bootstrap complete`. Am I doing smth wrong?

Reply:
Yeah, I mostly ran the `--help` commands locally a ton and didn't run the actual comamnds very often 😬. Fixed! [`a53a5e2` (#72318)](https://github.com/elastic/kibana/pull/72318/commits/a53a5e2a56c37ccf3849ba7bad5ca36e94369d8b)

Reply:
Make sure you bootstrap

Reply:
I still getting issue, it is now related to cli arguments:
```
node scripts/es_archiver.js save archive_name kibana_sample_data_ecommerce --dir test/functional/fixtures/es_archiver/
 info [save] Creating archive of ["archive_name","kibana_sample_data_ecommerce"]
ERROR UNHANDLED ERROR
ERROR Error: [index_not_found_exception] no such index [archive_name], with { resource.type="index_or_alias" & resource.id="archive_name" & index_uuid="_na_" & index="archive_name" }
          at respond (/Users/dzmitrylemechko/github/kibana/node_modules/elasticsearch/src/lib/transport.js:349:15)
          at checkRespForFailure (/Users/dzmitrylemechko/github/kibana/node_modules/elasticsearch/src/lib/transport.js:306:7)
          at HttpConnector.<anonymous> (/Users/dzmitrylemechko/github/kibana/node_modules/elasticsearch/src/lib/connectors/http.js:173:7)
          at IncomingMessage.wrapper (/Users/dzmitrylemechko/github/kibana/node_modules/lodash/lodash.js:4929:19)
          at IncomingMessage.emit (events.js:203:15)
          at endReadableNT (_stream_readable.js:1145:12)
          at process._tickCallback (internal/process/next_tick.js:63:19)
```

if I remove archive name, it works but folder is called `save`: 
```
➜  kibana git:(implement/es-archiver-package) ✗ node scripts/es_archiver.js save kibana_sample_data_ecommerce --dir test/functional/fixtures/es_archiver/
 info [save] Creating archive of ["kibana_sample_data_ecommerce"]
 info [save] Archived "kibana_sample_data_ecommerce"
 info [save] Archived 4675 docs from "kibana_sample_data_ecommerce"
```

Reply:
Thank you so much for actually testing this, I don't know why I didn't think I should actually verify this works... Thank you

Reply:
[`0780fe3` (#72318)](https://github.com/elastic/kibana/pull/72318/commits/0780fe3cf7f857bbcee5831656ad18d0eb835138)

![image](https://user-images.githubusercontent.com/1329312/87990142-43d39180-ca98-11ea-9200-ee025f652180.png)

Reply:
I tested both `save` and `load`, it works perfectly!